dlup/experimental_backends/openslide_backend.py

```diff
@@ -40,24 +42,41 @@ def __init__(self, filename: PathLike):
         try:
             mpp_x = float(self.properties[openslide.PROPERTY_NAME_MPP_X])
             mpp_y = float(self.properties[openslide.PROPERTY_NAME_MPP_Y])
+            self.spacing = (mpp_x, mpp_y)
 
         except KeyError:
-            raise UnsupportedSlideError(f"slide property mpp is not available.", str(filename))
+            pass
 
+    @property
+    def spacing(self) -> tuple[float, float] | None:
+        if not self._spacings:
+            return None
+        return self._spacings[0]
+
+    @spacing.setter
+    def spacing(self, value: tuple[float, float]) -> None:
+        if not isinstance(value, tuple) and len(value) != 2:
+            raise ValueError(f"`.spacing` has to be of the form (mpp_x, mpp_y).")
+
+        mpp_x, mpp_y = value
+        check_if_mpp_is_valid(mpp_x, mpp_y)
         mpp = np.array([mpp_y, mpp_x])
-        self._spacings = [tuple(mpp * downsample) for downsample in self.level_downsamples]
+        self._spacings = [cast(tuple[float, float], tuple(mpp * downsample)) for downsample in self.level_downsamples]
 
     @property
-    def magnification(self) -> float:
+    def magnification(self) -> float | None:
         """Returns the objective power at which the WSI was sampled."""
-        return int(self.properties[openslide.PROPERTY_NAME_OBJECTIVE_POWER])
+        value = self.properties.get(openslide.PROPERTY_NAME_OBJECTIVE_POWER, None)
+        if value is not None:
+            return int(value)
```

Comment:
The signature promises this function will return a float or None, but here any non-None values are being cast to int.

Reply:
Fixed